app/helpers/confirm.js

```diff
@@ -9,7 +9,12 @@ export default Helper.extend({
 
   compute(params) {
     return () => {
-      if (params.length >= 2) {
+      if (params.length >= 6) {
+        this.confirm.prompt('Are You Sure?', { 'denyText': params[2], 'approveText': params[3], 'denyColor': params[4], 'approveColor': params[5], 'extra': params[0] })
+          .then(() => {
+            params[1]();
+          });
+      } else if (params.length >= 2) {
```

Comment:
Can you explain please what is happening here. Looks very confusing

Reply:
`this.confirm.prompt('Are You Sure?', { 'denyText': params[2], 'approveText': params[3], 'denyColor': params[4], 'approveColor': params[5], 'extra': params[0] })`

the options are from service/confirm.js. In confirm-moda.js these are props (denyText, approveText...). in confirm modal helper i am using those and in component passing them.